File: src/main/java/bg/uniplovdiv/talkify/auth/user/model/UserSearchCriteria.java
package bg.uniplovdiv.talkify.auth.user.model;

import bg.uniplovdiv.talkify.common.encodedid.EncodedId;

public record UserSearchCriteria(
    String search,
    String username,
    String email,
    @EncodedId Long inChannelId,
    @EncodedId Long notInChannelId,
    Boolean onlyFriends,
    Boolean active) {}

File: src/main/frontend/src/models/user/UserSearchCriteria.ts
import Paginated from "../common/Paginated";

interface UserSearchCriteria extends Paginated {
  search?: string;
  username?: string;
  email?: string;
  inChannelId?: string;
  notInChannelId?: string;
  onlyFriends?: boolean;
  active?: boolean;
}

export default UserSearchCriteria;
